### ISSUE TITLE
How to configure algolia search for this project?

### ISSUE THREAD

**Issue by shipsywor** [ethereum/ethereum-org-website#2891]
**Is your feature request related to a problem? Please describe.**

I am failing to make algolia search work locally.

**Describe the solution you'd like**

Usage instructions on how to configure algolia search

**Describe alternatives you've considered**

I created `.env` and filled the values.

I noticed that `ALGOLIA_ADMIN_KEY` is never used in the entire project except `.env`

This leads me to believe if this is why my algolia search is not working?

I also changed the index name and url in `docSearchConfig.json` to the one I created on algolia site which is empty.

```
{
  "index_name": "MY_INDEX_NAME",
  "start_urls": [
    {
      "url": "http://localhost:8000/(?P<lang>.*?)/",
```

**Additional context**

I am a beginner.

Please guide on how to make the search work. What are the necessary configurations required, etc. Thank you.

**Comment by wackerow** [comment]
@shipsywor It should be the `GATSBY_ALGOLIA_SEARCH_KEY` .env variable that is used, but on that note...
@samajammin Can we safely share the public search API key (as Algolia suggests)? Or is this something we're keeping internal?

**Comment by shipsywor** [comment]
In the .env variable, I have inserted the appropriate values of algolia.

```
ALGOLIA_ADMIN_KEY=a925300caa..........................
DEFI_PULSE_API_KEY=keythisblank
ETHERSCAN_API_KEY=keythisblank
GATSBY_ALGOLIA_APP_ID=XAV.............
GATSBY_ALGOLIA_SEARCH_KEY=bf70bf29a88............................
GATSBY_GITHUB_TOKEN_READ_ONLY=keythisblank
```

Even after adding the .env variables, running `build` script and `start` script. The algolia search is not working.

**Additional Context**

In few other opensource gatsby projects, I got algolia working by adding the algolia admin key, search. app id values. But they were using `gatsby-plugin-algolia` which this ethereum gatsby project does not. Maybe that's why there's no usage of admin key.

**TLDR**
Is something not made opensource that I should in configuration to make algolia work?

Because just adding .env variables does not seem to work.

**Comment by minimalsm** [comment]
Related: work on #2764 is blocked by not having the ```GATSBY_ALGOLIA_SEARCH_KEY``` also.

**Comment by shipsywor** [comment]
> Related: work on #2764 is blocked by not having the ```GATSBY_ALGOLIA_SEARCH_KEY``` also.

For me, its searching nothing. On typing on search box, no search or search suggestions are performed.

**Comment by minimalsm** [comment]
> For me, its searching nothing. On typing on search box, no search or search suggestions are performed.

Yup same for me, this is what I meant.

**Comment by ryancreatescopy** [comment]
nudge @samajammin

**Comment by samajammin** [comment]
Hey folks, sorry for the delay on this @shipsywor! Thank you for offering to look into this.

Adding the following to your .env file should get this up & running:
```
GATSBY_ALGOLIA_APP_ID=A3QSF2Y6D5
GATSBY_ALGOLIA_SEARCH_KEY=43cc305190a93fa2f1ca7393ad21fb85
```
This will give you read access to our search index. If we discover this starts getting throttled with many requests I'll have to regenerate the Search API key but this should suit our needs for now.

You're correct about `ALGOLIA_ADMIN_KEY` - it's not used anywhere. I'll remove that from .env.example.

**Comment by shipsywor** [comment]
The use case for this project was to use this gatsby site (ethereum-org-website) you guys have developed as a template for the documentation of my opensource project.

My expectation was algolia would get somehow use the data in the markdown files to offer the search. 

For example, if I created a markdown file for some API documentation, algolia would automatically get all the text in that markdown file and use it to offer search.

However, you mentioned you use "search index".

So by using search index, how do you "transfer" all the text data in algolia search index on algolia site? 

By looking into algolia site, it says after "creating an index" under "import your records"

> Records are the information that you want to make searchable. You can push data via the API or by uploading records directly below.

Too much to ask, but could you please add some information regarding how you all guys push or upload data to algolia after creating an index in algolia site in README or in this issue so everyone could benefit those who simply want to use this opensource project as a template for their docs?

I am probably asking for too much considering this site is probably not primarily made for template purpose. Sorry for that.

**Comment by samajammin** [comment]
Gotcha. We use a GitHub Action that triggers the Algolia crawler to index our site whenever we merge into `master`:
- https://github.com/ethereum/ethereum-org-website/blob/dev/.github/workflows/docsearch-crawl.yml
- https://github.com/ethereum/ethereum-org-website/blob/dev/.github/workflows/docsearchConfig.json

This is not the only approach. I believe Algolia offers an option to index your site for free (you tell them the domain & they autoselect based on the HTML elements). I recommend reading up on Algolia's documentation. This [Gatsby guide](https://www.gatsbyjs.com/docs/adding-search-with-algolia/) might also be useful for you.

As this isn't related to ethereum.org development, I'm going to close this out.
